backup/queries_statistics.go

```diff
@@ -77,14 +77,16 @@ func GetAttributeStatistics(connectionPool *dbconn.DBConn, tables []Table, proce
 	for _, table := range tables {
 		tablenames = append(tablenames, table.FQN())
 	}
+	// In the query below we need to adjust the `staattnum` as some columns could have been dropped, but during database
+	// restore the tables will be restored from scratch, and their columns will have different attnums.
 	query := fmt.Sprintf(`
 	SELECT c.oid,
 		quote_ident(n.nspname) AS schema,
 		quote_ident(c.relname) AS table,
 		quote_ident(a.attname) AS attname,
 		quote_ident(t.typname) AS type,
 		s.starelid,
-		s.staattnum,
+		s.staattnum - (SELECT COUNT(1) FROM pg_attribute WHERE attrelid = c.oid AND attnum < s.staattnum AND attisdropped = TRUE) AS staattnum,
```

Comment:
Did you consider window function usage for this purpose? There should be lesser performance degradation because of ordering already exists.

Reply:
The window function will be able to operate only on the set of tuples in `FROM` clause, that doesn't have info about dropped columns (because it is a join with `pg_statistic` where rows for dropped columns are removed). So I do not see a way how to leverage window function here. Do you?

Reply:
Note: it is no more relevant, the solution was reconsidered.

Reply:
Of course, join with pg_attribute was considered. In my opinion, it's more performant than sub query